### PR TITLE
test: add test for fibonacci() function

### PR DESCRIPTION
**Pull request by fzxutar**
#### Description of Change
- Added a test function for the fibonacci() function.

#### Checklist

- [x] Added description of change
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 
A simple test function was added to test the fibonacci() function to make sure that the results are correct.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1023"></a>

### CODE REVIEW

**Comment by kvedala** [comment]
It seems that the error is from the existing code and not in your contribution. 
As the error states, the function `fibonacci()` is returning an `unsigned int` while its definition for return type is `int`. 
This can be rectified by changing the function return type on line 17:
```cpp
unsigned int fibonacci(unsigned int n) {
```
image

File: math/fibonacci.cpp [hunk 1]
```diff
@@ -24,8 +24,39 @@ int fibonacci(unsigned int n) {
     return fibonacci(n - 1) + fibonacci(n - 2);
 }
 
+/**
+ * Function for testing the fibonacci() function with a few
+ * test cases and assert statement.
+ */
+void test() {
```
Comment (kvedala) [inline math/fibonacci.cpp:31]:
```suggestion
static void test() {
```
This is to specify that the function is only for this particular file and not to be globally visible.

File: math/fibonacci.cpp [hunk 2]
```diff
@@ -24,8 +24,39 @@ int fibonacci(unsigned int n) {
     return fibonacci(n - 1) + fibonacci(n - 2);
 }
 
+/**
+ * Function for testing the fibonacci() function with a few
+ * test cases and assert statement.
+ */
```
Comment (kvedala) [inline math/fibonacci.cpp:30]:
```suggestion
  * @returns `void`
  */
```
This is needed for the documentation generator to recognize `test()` as a function.

Comment (fzxutar) [reply math/fibonacci.cpp]: Thanks for the suggestions. I fixed the errors and now all checks have passed.

**Pull request merged**